### PR TITLE
[LM - 8246] Load Testing Framework Initialisation

### PR DESCRIPTION
**Pull request by Mahir-G2**
Reviewer's Checklist:
- [x] Jira ticket attached 
- [x] Pull request title has jira ticket id prefix
- [ ] Description added for code changes
- [ ] API documentation added/updated (attach link if applicable)
- [ ] Deployment instructions attached, if any

Fill the required checkboxes, ignore if N/A (not applicable)
Ticket
- https://likemindscommunity.atlassian.net/browse/LM-8246

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,35 +1,187 @@
...
+                }
+                expression {
+                    return params.action == 'apply'
+                }
+            }
+            steps {
+                echo 'Kettle Selected'
+                script {
+                    dir('likeminds-authentication') {
+                        git branch: "${kettle_branch}", credentialsId: 'df5b81c3-2bfe-4938-a421-5f55f996e76a', url: 'https://github.com/NateshR/LikeMinds-Authentication/'
```
Comment (yadavKetan) [inline Jenkinsfile:24]: what is this credentialId ?

File: Jenkinsfile [hunk 2]
```diff
@@ -1,35 +1,187 @@
...
+                    -var 'caravan_rabbitmq_password=${caravan_rabbitmq_password}'
+                    """
                 }
             }
         }
     }
-
-    post {
-        always {
-            cleanWs()
```
Comment (yadavKetan) [inline Jenkinsfile:32]: should clean workspace, why is this removed ?

File: README.md
Comment (yadavKetan) [inline README.md:1]: Add a basic intro to the repository.
What is this project.
How to setup locally.
how to runs docs can be attached.

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,51 @@
+terraform {
+  required_providers {
+    google = {
+      source  = "hashicorp/google"
+      version = "3.52.0"
+    }
+    kubernetes = {
+      source  = "hashicorp/kubernetes"
+      version = ">= 2.6.2"
```
Comment (yadavKetan) [inline main.tf:9]: should use the same version as our existing GKE clusters, and lock the cluster version.

File: main.tf [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+# https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/client_config
+# This fetches a new token, which will expire in 1 hour.
+data "google_client_config" "default" {}
+
+data "google_container_cluster" "likeminds-load-testing-autopilot-cluster" {
+  name     = var.cluster_name
+  location = var.region
+}
+
+provider "kubernetes" {
```
Comment (yadavKetan) [inline main.tf:24]: can we not use static string var names here.

File: provider.tf [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# Retrieve GKE cluster information
+provider "google" {
+  project = var.project_id
+  region  = var.region
+}
```
Comment (yadavKetan) [inline provider.tf:5]: add newline at EOF

File: swarm.tf [hunk 6]
```diff
@@ -0,0 +1,131 @@
+resource "kubernetes_deployment" "swarm-load" {
```
Comment (yadavKetan) [inline swarm.tf:1]: can we use static string var names for all these strings ?

File: swarm.tf [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+      "cloud.google.com/neg": "{\"ingress\": true}"
+    }
+  }
+
+  spec {
+    selector = {
+      app = var.swarm_app_name
+    }
+    port {
+      name        = "http"
```
Comment (yadavKetan) [inline swarm.tf:65]: use variables to define this at use here, in place of strings and integers

File: Jenkinsfile (hunk 1 above)
Comment (Mahir-G2) [reply Jenkinsfile:24]: This is the ID for your account that is added on the jenkins dashboard.

File: Jenkinsfile (hunk 2 above)
Comment (Mahir-G2) [reply Jenkinsfile:32]: Terraform creates a state file, if we clean the workspace, that state file also gets removed, which won't maintain the resource's states.

File: main.tf (hunk 3 above)
Comment (Mahir-G2) [reply main.tf:9]: This is not the Kubernetes version, this is the version for Hasicorp's Kubernetes provider. We can lock it to the current version, but it is not dependent on our Kubernetes cluster version.

File: main.tf (hunk 4 above)
Comment (Mahir-G2) [reply main.tf:24]: No, terraform doesn't give support for resource naming via variables. 
Ref:- https://stackoverflow.com/questions/38619691/referring-to-resources-named-with-variables-in-terraform

File: swarm.tf (hunk 6 above)
Comment (Mahir-G2) [reply swarm.tf:1]: No, terraform doesn't give support for resource naming via variables.
Ref:- https://stackoverflow.com/questions/38619691/referring-to-resources-named-with-variables-in-terraform

File: provider.tf (hunk 5 above)
Comment (Mahir-G2) [reply provider.tf]: done

File: swarm.tf (hunk 7 above)
Comment (Mahir-G2) [reply swarm.tf]: done

File: README.md
Comment (Mahir-G2) [reply README.md:1]: Updated

File: Jenkinsfile (hunk 1 above)
Comment (yadavKetan) [reply Jenkinsfile:24]: for the git credentials, as in to access git repository via token ?

File: main.tf (hunk 3 above)
Comment (yadavKetan) [reply main.tf]: got it, lock it to the current version then, as an update that is not backward compatible might break.

**Review by yadavKetan — APPROVED** [review]
We should maintain an application version for this as well. add the var to an appropriate location.
version - https://likemindscommunity.atlassian.net/projects/LM/versions/10386/tab/release-report-all-issues

**Comment by Mahir-G2** [comment]
> We should maintain an application version for this as well. add the var to an appropriate location. version - https://likemindscommunity.atlassian.net/projects/LM/versions/10386/tab/release-report-all-issues

- Updated the app version in the main.tf file
- yes, the credentialsId is for git credentials, to access the git repo
- locked the current version

File: main.tf (hunk 3 above)
Comment (yadavKetan) [reply main.tf]: still uses version >= , is this locked ?

Comment (Mahir-G2) [reply main.tf]: updated

**Review by yadavKetan — APPROVED** [review]

**Review by yadavKetan — APPROVED** [review]
